examples/text-classification/run_xnli.py

```diff
@@ -16,611 +16,315 @@
-        do_lower_case=args.do_lower_case,
-        cache_dir=args.cache_dir,
+        model_args.tokenizer_name if model_args.tokenizer_name else model_args.model_name_or_path,
+        do_lower_case=model_args.do_lower_case,
+        cache_dir=model_args.cache_dir,
+        use_fast=model_args.use_fast_tokenizer,
+        revision=model_args.model_revision,
+        use_auth_token=True if model_args.use_auth_token else None,
     )
     model = AutoModelForSequenceClassification.from_pretrained(
-        args.model_name_or_path,
-        from_tf=bool(".ckpt" in args.model_name_or_path),
+        model_args.model_name_or_path,
+        from_tf=bool(".ckpt" in model_args.model_name_or_path),
         config=config,
-        cache_dir=args.cache_dir,
+        cache_dir=model_args.cache_dir,
+        revision=model_args.model_revision,
+        use_auth_token=True if model_args.use_auth_token else None,
     )
 
-    if args.local_rank == 0:
-        torch.distributed.barrier()  # Make sure only the first process in distributed training will download model & vocab
+    # Preprocessing the datasets
+    sentence1_key, sentence2_key = task_to_keys[task_name]
+
+    # Padding strategy
+    if data_args.pad_to_max_length:
+        padding = "max_length"
+    else:
+        # We will pad later, dynamically at batch creation, to the max sequence length in each batch
+        padding = False
 
-    model.to(args.device)
+    def preprocess_function(examples):
+        # Tokenize the texts
+        args = (
+            (examples[sentence1_key],) if sentence2_key is None else (examples[sentence1_key], examples[sentence2_key])
+        )
+        result = tokenizer(*args, padding=padding, max_length=data_args.max_seq_length, truncation=True)
+        return result
+
+    train_dataset = train_dataset.map(
+        preprocess_function, batched=True, load_from_cache_file=not data_args.overwrite_cache
+    )
+    eval_dataset = eval_dataset.map(
+        preprocess_function, batched=True, load_from_cache_file=not data_args.overwrite_cache
+    )
 
-    logger.info("Training/evaluation parameters %s", args)
+    # Log a few random samples from the training set:
+    for index in random.sample(range(len(train_dataset)), 3):
+        logger.info(f"Sample {index} of the training set: {train_dataset[index]}.")
+
+    # Get the metric function
+    metric = load_metric(task_name)
+
+    # You can define your custom compute_metrics function. It takes an `EvalPrediction` object (a namedtuple with a
+    # predictions and label_ids field) and has to return a dictionary string to float.
+    def compute_metrics(p: EvalPrediction):
+        preds = p.predictions[0] if isinstance(p.predictions, tuple) else p.predictions
+        preds = np.argmax(preds, axis=1)
+        result = metric.compute(predictions=preds, references=p.label_ids)
+        if len(result) > 1:
+            result["combined_score"] = np.mean(list(result.values())).item()
```

Comment:
What is the metric for xnli? Depending on it we can either remove the two lines (if there is just one) or remove the test.

Reply:
The metric for xnli is just the simple accuracy,
```
>>> from datasets import load_metric
>>> metric = load_metric("xnli")
>>> metric.compute
<bound method compute of Metric(name: "xnli", features: {'predictions': Value(dtype='int64', id=None), 'references': Value(dtype='int64', id=None)}, usage: """
Computes XNLI score which is just simple accuracy.
Args:
    predictions: Predicted labels.
    references: Ground truth labels.
Returns:
    'accuracy': accuracy
```
hence I've removed the test for `len(result)>1` and amended `compute_metrics()` to

https://github.com/huggingface/transformers/blob/0f6858f34ff5b56a451f893548269904f3efb751/examples/text-classification/run_xnli.py#L260-L263